src/runtime/store/index.ts

```diff
@@ -99,6 +99,9 @@ export function writable<T>(value?: T, start: StartStopNotifier<T> = noop): Writ
 		return () => {
 			subscribers.delete(subscriber);
 			if (subscribers.size === 0) {
+				if (stop === null) {
+					throw new Error('unsubscribe cannot be called twice');
+				}
 				stop();
 				stop = null;
```

Comment:
I think the more robust approach (which also needs less code) would be to make `stop` a noop
```suggestion
				stop();
				stop = noop;
```

Reply:
Ah, yeah. I thought that error was deliberate. Of course it's much better to avoid the error completely and allow the unsubscribe function to be called twice.

(It might be nice to issue a console warning, though. In our case, it was an actual bug in our code.)